### DISCUSSION TITLE
Consider allowing dynamic components in comments and texts

### DISCUSSION THREAD

**Discussion by tsmyre** [ManifoldScholar/manifold#3450] [Ideas]
We've talked in person and on Slack about allowing markdown, embeds, and perhaps dynamic components in annotations, comments, and maybe even in texts. The two repos below have come up specifically, and it may align with the work slated for v4.0. I note them here so we have a record we can easily reference and refer back to.

- [Idyll Toolkit for Interactive Text](https://idyll-lang.org/)
- [Slate Rich-Text Editor](https://docs.slatejs.org/)

**Comment by mkgold** [comment]
awesome

**Comment by SMaxOwok** [comment]
I believe this is a dupe of #1721, unless there is more functionality included here.

**Comment by tsmyre** [comment]
@SMaxOwok It looks like #1727 is also closely aligned. Neither of those mention [Idyll](https://idyll-lang.org/), however.

**Comment by zdavis** [comment]
Given our other priorities, I don't see us tackling the Idyll integration in the near-term. I think this should be flagged as funding needed and closed for the time being, esp. because we're already addressing the RTE aspect of this in #1721, which is more actionable at this time.

**Comment by bri-caszatt** [comment]
I would love to see more functionality in the annotations. My next project in one of my classes is to create an annotated edition of a text, and we've picked Dracula. I'm pushing for us to use Manifold as it offers so many great features, but we're planning on having image-heavy annotations. Our annotations will look at the history that inspired the story, as well as its cultural impact, the latter of which includes a lot of gifs and memes. We also plan to explore the text with Python and NLTK, and want to include data visualizations in our annotations. Right now we can only include these as anchored digital objects, which will create a disjointed reader experience. We'd ideally like our non-text annotations to carry as much visual weight as the text ones, and for all of our annotations to live in the same area.

Additionally, I've used Hypothesis for group reading and annotating in other classes. It's so much more fun to be able to include images, gifs, and videos in the annotations, and I think it resulted in a high level of engagement with the course materials.

I really hope you get funding to implement this functionality for annotations.
